reposerver/repository/repository.go

```diff
@@ -103,6 +102,27 @@ func (s *Service) GenerateManifest(c context.Context, q *ManifestRequest) (*Mani
 	}, nil
 }
 
+func (s *Service) setAppLabels(target *unstructured.Unstructured, appName string) error {
+	labels := target.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[common.LabelApplicationName] = appName
+	target.SetLabels(labels)
+	// special case for deployment: make sure that derived replicaset and pod has application label
+	if target.GetKind() == "Deployment" {
```

Comment:
We should also do this for StatefulSet

Reply:
We will need to make mode changes for StatefulSet and Services. I plan to work on it in separate PR.